android/src/main/java/com/rivereactnative/RiveReactNativeView.kt

```diff
@@ -551,16 +559,44 @@ class RiveReactNativeView(private val context: ThemedReactContext) : FrameLayout
   }
 
   fun setAnimationName(animationName: String) {
+    if (this.animationName == animationName) return
     this.animationName = animationName
     shouldBeReloaded = true
   }
 
   fun setReferencedAssets(referencedAssets: ReadableMap?) {
-    this.referencedAssets = referencedAssets;
-    shouldBeReloaded = true;
+    if (this.referencedAssets?.toMap() == referencedAssets?.toMap()) return
+
+    val previousReferencedAssets = this.referencedAssets
+    this.referencedAssets = referencedAssets
+
+    if (previousReferencedAssets == null || referencedAssets == null) {
+      return
+    }
+
+    val previousKeys = previousReferencedAssets.keysList();
+    val newKeys = referencedAssets.keysList()
+
+    if (previousKeys.toSet() != newKeys.toSet()) {
+      shouldBeReloaded = true
+      return
+    }
+
+    for (key in newKeys) {
+      val previousValue = previousReferencedAssets.getMap(key)
+      val newValue = referencedAssets.getMap(key)
+      if (previousValue?.toMap() != newValue?.toMap()) {
+        val source = newValue?.getMap("source")
+        val asset = assetStore?.cachedFileAssets?.get(key)
+        if (source != null && asset != null) {
+          loadAsset(source, asset)
+        }
+      }
+    }
```

Comment:
For your consideration. It reads better in my opinion, but some people prefer iterative over collection operations.
```suggestion
    newKeys
        .filter { key ->
            previousReferencedAssets.getMap(key)?.toMap() != referencedAssets.getMap(key)?.toMap()
        }
        .map { key ->
            referencedAssets.getMap(key)?.getMap("source") to assetStore?.cachedFileAssets?.get(key)
        }
        .filter { (source, asset) -> source != null && asset != null }
        .forEach { (source, asset) -> loadAsset(source!!, asset!!) }
```